ambry-quota/src/main/java/com/github/ambry/quota/capacityunit/UnlimitedQuotaSource.java

```diff
@@ -17,15 +17,31 @@
 import com.github.ambry.quota.QuotaName;
 import com.github.ambry.quota.QuotaResource;
 import com.github.ambry.quota.QuotaSource;
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.List;
 
 
 /**
  * An implementation of {@link QuotaSource} that always returns a max value for {@link Quota} for all resources.
  */
 public class UnlimitedQuotaSource implements QuotaSource {
+  private final List<QuotaResource> quotaResourceList = new ArrayList<>();
 
   @Override
   public Quota getQuota(QuotaResource quotaResource, QuotaName quotaName) {
     return new Quota<>(quotaName, Long.MAX_VALUE, quotaResource);
   }
+
+  @Override
+  public void updateNewQuotaResources(Collection<QuotaResource> quotaResources) {
+    quotaResourceList.addAll(quotaResources);
```

Comment:
given the fact the "updated" account (not newly created) will also be passed to this method, we might see same quota resource multiple times in the list. One way to fix it is to change the list to set.